client/my-sites/upgrades/cart/cart-trial-ad.jsx

```diff
@@ -2,35 +2,39 @@
  * External dependencies
  */
 import React from 'react';
+import { connect } from 'react-redux';
+import page from 'page';
 import i18n from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import CartAd from './cart-ad';
 import { cartItems } from 'lib/cart-values';
-import { addCurrentPlanToCartAndRedirect, getCurrentPlan, getDayOfTrial } from 'lib/plans';
+import { getDayOfTrial } from 'lib/plans';
+import { addItem } from 'lib/upgrades/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteSlug } from 'state/sites/selectors';
+import { getCurrentPlan, isRequestingSitePlans } from 'state/sites/plans/selectors';
 
 const CartTrialAd = React.createClass( {
 	propTypes: {
 		cart: React.PropTypes.object.isRequired,
-		sitePlans: React.PropTypes.object.isRequired,
-		selectedSite: React.PropTypes.oneOfType( [
-			React.PropTypes.object,
-			React.PropTypes.bool
-		] ).isRequired
+		// connected props
+		currentPlan: React.PropTypes.object,
+		isRequestingSitePlans: React.PropTypes.bool,
+		selectedSiteSlug: React.PropTypes.string
 	},
 
 	addPlanAndRedirect( event ) {
 		event.preventDefault();
-
-		addCurrentPlanToCartAndRedirect( this.props.sitePlans, this.props.selectedSite );
+		addItem( cartItems.planItem( this.props.currentPlan.productSlug ) );
```

Comment:
Optional we could connect pass through the productSlug and productName in connect.